spec/services/workflow_errors_reporter_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'rails_helper'
+
+RSpec.describe WorkflowErrorsReporter do
+  let(:put) { 'https://sul-lyberservices-test.stanford.edu/workflow/dor/objects/druid:jj925bx9565/workflows/preservationWF/moab-valid' }
+  let(:headers) { { 'Content-Type' => 'application/xml' } }
+  let(:result_array) do
+    { 13 => "Invalid moab, validation error...ential version directories." }
+  end
+  let(:body) { "<process name='moab-valid status='error' errorMessage='{13=>\"Invalid moab, validation error...ential version directories.\"}'/>" }
+
+  describe '.update_workflow with 204' do
```

Comment:
I would tweak this slightly (and no, i do not love infinite nesting)

```ruby
context '.update_workflow' do
  it '204 response' do
     # code within the before
     # actual test code showing you expect message
  end

  it '400 response' do
   # code within THAT before
   # actual test code
  end
 
  it 'invalid blah blah url' do
    etc
  end
end
```

Reply:
i need each of the `it` blocks to have a `describe` block so i can use the `before do...` block

Reply:
you don't need a `before do` block if you just put that code inside the `it` block.  That is, if the setup is only for a single test, it can be part of a single test.

If there is setup applying to multiple tests, it can be a `before` block inside the context block.  `context` is basically equal to `describe` anyway.

Reply:
oh right, i knew that..